### PR TITLE
misc small patches

### PR DESCRIPTION
**Pull request by etbe**

### CODE REVIEW

File: policy/modules/admin/bootloader.if [hunk 1]
```diff
@@ -122,6 +122,62 @@ interface(`bootloader_rw_tmp_files',`
...
+## <summary>
+##      read bootloader link files under /tmp
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>
+## </param>
+#
+interface(`bootloader_read_tmp_lnk_files',`
```
Comment (pebenito) [inline policy/modules/admin/bootloader.if:173]: bootloader_read_tmp_symlinks()

File: policy/modules/admin/bootloader.te [hunk 2]
```diff
@@ -41,10 +41,11 @@ dev_node(bootloader_tmp_t)
 # bootloader local policy
 #
 
-allow bootloader_t self:capability { chown dac_override dac_read_search fsetid mknod setgid sys_admin sys_rawio };
+allow bootloader_t self:capability { chown dac_override dac_read_search fsetid sys_chroot mknod setgid sys_admin sys_rawio };
 dontaudit bootloader_t self:capability sys_resource;
-allow bootloader_t self:process { signal_perms execmem };
+allow bootloader_t self:process { signal_perms execmem getsched setfscreate };
 allow bootloader_t self:fifo_file rw_fifo_file_perms;
+allow bootloader_t self:netlink_selinux_socket connected_socket_perms;
```
Comment (pebenito) [inline policy/modules/admin/bootloader.te:48]: I need more information on the bootloader changes. There are too many and there is no explanation in this PR. I didn't get any results of `grep -i selinux` in the grub source code. The only result was in `tests/util/grub-fs-tester.in` for mount options.

File: policy/modules/admin/brctl.te [hunk 3]
```diff
@@ -17,12 +17,13 @@ role brctl_roles types brctl_t;
 # Local policy
 #
 
-allow brctl_t self:capability net_admin;
+allow brctl_t self:capability { net_admin };
```
Comment (pebenito) [inline policy/modules/admin/brctl.te:20]: Unnecessary change.

File: policy/modules/admin/brctl.te [hunk 4]
```diff
@@ -17,12 +17,13 @@ role brctl_roles types brctl_t;
...
 allow brctl_t self:fifo_file rw_fifo_file_perms;
 allow brctl_t self:unix_stream_socket create_stream_socket_perms;
 allow brctl_t self:unix_dgram_socket create_socket_perms;
 allow brctl_t self:tcp_socket create_socket_perms;
 
+kernel_load_module(brctl_t)
```
Comment (pebenito) [inline policy/modules/admin/brctl.te:26]: Can this be dontaudited instead? I would hope the below module load request would be sufficient.

File: policy/modules/admin/logrotate.te [hunk 5]
```diff
@@ -117,6 +117,8 @@ init_dbus_chat(logrotate_t)
 init_stream_connect(logrotate_t)
 init_manage_all_units(logrotate_t)
 
+libs_exec_lib_files(logrotate_t)
```
Comment (pebenito) [inline policy/modules/admin/logrotate.te:120]: Seems odd, possibly mislabeled file(s)?

File: policy/modules/apps/games.te [hunk 6]
```diff
@@ -128,6 +132,8 @@ corenet_tcp_bind_generic_port(games_t)
 corenet_sendrecv_generic_client_packets(games_t)
 corenet_tcp_connect_generic_port(games_t)
 
+corenet_udp_bind_generic_node(games_t)
```
Comment (pebenito) [inline policy/modules/apps/games.te:135]: This seems out of place without any port bindings.

File: policy/modules/apps/games.te [hunk 7]
```diff
@@ -158,6 +167,7 @@ userdom_manage_user_tmp_dirs(games_t)
 userdom_manage_user_tmp_files(games_t)
 userdom_manage_user_tmp_symlinks(games_t)
 userdom_manage_user_tmp_sockets(games_t)
+userdom_use_user_ptys(games_t)
```
Comment (pebenito) [inline policy/modules/apps/games.te:170]: `userdom_use_inherited_user_terminals` is probably a better fit.

File: policy/modules/apps/gpg.te [hunk 8]
```diff
@@ -138,6 +138,7 @@ miscfiles_read_localization(gpg_t)
 miscfiles_read_generic_certs(gpg_t)
 
 userdom_use_user_terminals(gpg_t)
+userdom_user_home_dir_filetrans_user_home_content(gpg_t, file)
```
Comment (pebenito) [inline policy/modules/apps/gpg.te:141]: This should go in the `gpg_manage_all_user_content` block.

File: policy/modules/kernel/filesystem.if [hunk 9]
```diff
@@ -6732,3 +6732,21 @@ interface(`fs_unconfined',`
...
+##	</summary>
+## </param>
+#
+interface(`fs_search_bpf',`
+	gen_require(`
+		type bpf_t;
+	')
+
+	allow $1 bpf_t:dir search;
+')
```
Comment (pebenito) [inline policy/modules/kernel/filesystem.if:6752]: This should move up between `fs_mount_bpf` and `fs_create_bpf_dirs`.

File: policy/modules/roles/sysadm.te [hunk 10]
```diff
@@ -54,6 +56,7 @@ ubac_fd_exempt(sysadm_t)
 
 init_exec(sysadm_t)
 init_admin(sysadm_t)
+init_rw_stream_sockets(sysadm_t)
```
Comment (pebenito) [inline policy/modules/roles/sysadm.te:59]: What function(s) is this related to?

File: policy/modules/system/authlogin.te [hunk 11]
```diff
@@ -439,6 +440,8 @@ domain_use_interactive_fds(utempter_t)
 
 logging_search_logs(utempter_t)
 
+term_use_ptmx(utempter_t)
```
Comment (pebenito) [inline policy/modules/system/authlogin.te:443]: This means a pty is created, but there is no new pty type.

File: policy/modules/system/init.if [hunk 12]
```diff
@@ -3685,6 +3685,24 @@ interface(`init_reload_all_units',`
...
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`init_getattr_all_units',`
+	gen_require(`
+		attribute systemdunit;
+	')
+
+	allow $1 systemdunit:file getattr;
```
Comment (pebenito) [inline policy/modules/system/init.if:3703]: Please add dir and symlink access similar to the following interface.

File: policy/modules/system/init.te [hunk 13]
```diff
@@ -272,7 +272,6 @@ ifdef(`init_systemd',`
 	allow init_t self:udp_socket create_socket_perms;
 	allow init_t self:netlink_route_socket create_netlink_socket_perms;
 	allow init_t initrc_t:unix_dgram_socket create_socket_perms;
-	allow init_t self:capability2 audit_read;
```
Comment (pebenito) [inline policy/modules/system/init.te:275]: Need reasoning.

File: policy/modules/system/lvm.if [hunk 14]
```diff
@@ -211,3 +211,22 @@ interface(`lvm_admin',`
...
+## </param>
+#
+interface(`lvm_manage_metadata',`
+	gen_require(`
+		type lvm_metadata_t;
+	')
+
+	allow $1 lvm_metadata_t:dir manage_dir_perms;
+	allow $1 lvm_metadata_t:file manage_file_perms;
+')
```
Comment (pebenito) [inline policy/modules/system/lvm.if:232]: Please move above `lvm_admin()`

File: policy/modules/system/modutils.te [hunk 15]
```diff
@@ -33,7 +33,8 @@ ifdef(`init_systemd',`
 # insmod local policy
 #
 
-allow kmod_t self:capability { dac_override net_raw sys_nice sys_tty_config };
+allow kmod_t self:capability { dac_override dac_read_search net_raw sys_nice sys_tty_config };
+allow kmod_t self:lockdown confidentiality;
```
Comment (pebenito) [inline policy/modules/system/modutils.te:37]: `lockdown` was removed. Please drop.

File: policy/modules/system/modutils.te [hunk 16]
```diff
@@ -45,6 +46,8 @@ allow kmod_t self:key write;
 # Read module config and dependency information
 list_dirs_pattern(kmod_t, modules_conf_t, modules_conf_t)
 read_files_pattern(kmod_t, modules_conf_t, modules_conf_t)
+allow kmod_t modules_conf_t:lnk_file read_lnk_file_perms;
```
Comment (pebenito) [inline policy/modules/system/modutils.te:49]: Please use `read_lnk_files_pattern` for uniformity with the previous lines.

File: policy/modules/system/modutils.te [hunk 17]
```diff
@@ -114,6 +117,7 @@ miscfiles_read_localization(kmod_t)
 
 seutil_read_file_contexts(kmod_t)
 
+term_use_unallocated_ttys(kmod_t)
```
Comment (pebenito) [inline policy/modules/system/modutils.te:120]: What is the use case?

File: policy/modules/system/mount.te [hunk 18]
```diff
@@ -102,6 +102,7 @@ files_dontaudit_setattr_all_mountpoints(mount_t)
 fs_getattr_all_fs(mount_t)
 fs_getattr_all_dirs(mount_t)
 fs_mount_all_fs(mount_t)
+fs_manage_tmpfs_dirs(mount_t)
```
Comment (pebenito) [inline policy/modules/system/mount.te:105]: What is the use case?

**Review by pebenito — CHANGES_REQUESTED** [review]
This PR is too big to have no description. Ideally, each patch would be an individual commit with a description of the change in the commit message. That way everyone knows the reasoning, now and into the future.

File: policy/modules/admin/bootloader.if (hunk 1 above)
Comment (etbe) [reply policy/modules/admin/bootloader.if:173]: ok

File: policy/modules/admin/bootloader.te (hunk 2 above)
Comment (etbe) [reply policy/modules/admin/bootloader.te:48]: I'll back that out and resubmit when I find a good explanation

File: policy/modules/admin/brctl.te (hunk 3 above)
Comment (etbe) [reply policy/modules/admin/brctl.te:20]: ok

File: policy/modules/admin/brctl.te (hunk 4 above)
Comment (etbe) [reply policy/modules/admin/brctl.te:26]: I'll try that

File: policy/modules/admin/logrotate.te (hunk 5 above)
Comment (etbe) [reply policy/modules/admin/logrotate.te:120]: I'll revert and try again later

File: policy/modules/system/lvm.if (hunk 14 above)
Comment (etbe) [reply policy/modules/system/lvm.if:232]: done

File: policy/modules/system/modutils.te (hunk 15 above)
Comment (etbe) [reply policy/modules/system/modutils.te:37]: ok

File: policy/modules/system/modutils.te (hunk 16 above)
Comment (etbe) [reply policy/modules/system/modutils.te:49]: ok

File: policy/modules/system/mount.te (hunk 18 above)
Comment (etbe) [reply policy/modules/system/mount.te:105]: I'll revert and get more data

File: policy/modules/system/modutils.te (hunk 17 above)
Comment (etbe) [reply policy/modules/system/modutils.te:119]: sysadmin running modprobe from sulogin.
